unyt/array.py

```diff
@@ -265,7 +265,12 @@ def _coerce_iterable_units(input_object, registry=None):
                 return unyt_array(np.array(ret), ff, registry=registry)
             # This will create a copy of the data in the iterable.
             return unyt_array(np.array(input_object), ff, registry=registry)
-    return np.asarray(input_object)
+        ret = np.asarray(input_object)
+    else:
+        ret = np.asarray(input_object)
+    if ret.dtype.char in ("S", "U", "a", "O", "M", "m", "b"):
```

Comment:
this tuple should probably be a module-level constant for slightly better performance